src/yunohost/domain.py

```diff
@@ -448,21 +448,16 @@ def _build_dns_conf(domain, ttl=3600):
 
     if ipv6:
         basic.append(["@", ttl, "AAAA", ipv6])
+    elif include_empty_AAAA_if_no_ipv6:
+        basic.append(["@", ttl, "AAAA", None])
 
     #########
     # Email #
     #########
 
-    spf_record = '"v=spf1 a mx'
-    if ipv4:
-        spf_record += ' ip4:{ip4}'.format(ip4=ipv4)
-    if ipv6:
-        spf_record += ' ip6:{ip6}'.format(ip6=ipv6)
-    spf_record += ' -all"'
-
     mail = [
         ["@", ttl, "MX", "10 %s." % domain],
-        ["@", ttl, "TXT", spf_record],
+        ["@", ttl, "TXT", "v=spf1 a mx -all"],
```

Comment:
Why this change?

Reply:
ljf explained to me that in fact, having `ip4:11.22.33.44` or `ip6:11:22:33..`  is in fact redudant with the `a` and `mx` in this record. `a` and `mx` meaning is to declare the IP being A, AAAA and MX records as legitimate to send emails. So it's redundant in the context of our recommendation to re-add `ip4` etc ...

 (And as a side note it also makes it possibly more difficult for people with dynamic IP if the record depends on the IP, they have to update another record instead of just the A / AAAArecord)

You could however, in more complex setup, have other machines that you want to allow to send email (typically if you have a Gandi / OVH / ... whatever mailbox on the same domain) and in such case you might end up wiling to add the IP of those extra machines with `ip4:` and `ip6:`. Which is why later in diagnosis I ignore the parts about `ip4` and `ip6`, also to not raise a warning/error for people that configured with the old recommendation.